feat: add UI configuration, automation, cert usage

liquidat opened the pull request:
- Add information where to enter configuration at all
- Rewrite configuration guide to be UI first
- Add guide on how to renew certificates
- Add details and example on how to use the certificates in the Home Assistant default installation
- Resort DNS provider data: now all in one place and not spread, also long lists hidden behind details

This time also signing the CLA. Sorry about that. See #2888 for the same attempt.

sairon commented on letsencrypt/DOCS.md:
```diff
@@ -95,6 +114,41 @@ inwx_username: ''
+
+There are multiple ways how the add-on can be started to check/renew the certificates. One way to automate the certificate renewal it to use an automation, starting the add-on for example every night:
+
+1. Navigate in your Home Assistant frontend to **Settings** -> **Automations & Scenes**
+2. Click the button **CREATE AUTOMATION**
+3. Pick **Start with empty automation**
+4. Click **ADD TRIGGER**, pick **Time** and enter any time at the day
+5. Click **ADD ACTION**, pick **Call Service**, pick **Home Assistant Supervisor: Restart add-on.**, pick the add-on **Let's Encrypt**
+6. Click **SAVE**
```
Instead of this block, I will prefer to be a bit more vague - just add a link to [HA docs](https://www.home-assistant.io/docs/automation/editor/) and instruct people to use the "Restart add-on" action on specific time. Otherwise it adds some documentation maintenance burden to these docs.

liquidat replied:
I updated the corresponding part, please take a look.

sairon commented on letsencrypt/DOCS.md:
````diff
@@ -23,26 +29,39 @@ To use this add-on, you have two options on how to get your certificate:
-dns:
-  provider: ""
+```txt
+dns-azure
+dns-cloudflare
+dns-cloudxns
+dns-digitalocean
+dns-directadmin
+dns-dnsimple
+dns-dnsmadeeasy
+dns-gehirn
+dns-google
+dns-hetzner
+dns-linode
+dns-luadns
+dns-njalla
+dns-nsone
+dns-ovh
+dns-rfc2136
+dns-route53
+dns-sakuracloud
+dns-netcup
+dns-gandi
+dns-transip
+dns-inwx
````
Change it to the (so far) up-to-date list:
```suggestion
dns-azure
dns-cloudflare
dns-desec
dns-digitalocean
dns-directadmin
dns-dnsimple
dns-dnsmadeeasy
dns-duckdns
dns-dreamhost
dns-gehirn
dns-google
dns-hetzner
dns-infomaniak
dns-linode
dns-luadns
dns-njalla
dns-nsone
dns-ovh
dns-rfc2136
dns-route53
dns-sakuracloud
dns-namecheap
dns-netcup
dns-gandi
dns-transip
dns-inwx
dns-porkbun
```

liquidat replied:
Done.

sairon reviewed (changes requested):
Sorry it's been taking so long to review. In my view it makes sense to add few more hints about the setup. Could you please just update the PR to fix conflicts with the current `master` branch and see my comment below?

liquidat commented:
> Sorry it's been taking so long to review. In my view it makes sense to add few more hints about the setup. Could you please just update the PR to fix conflicts with the current `master` branch and see my comment below?

Hi, don't you worry, I am happy that someone has the time to look at this rather small change.
I  fixed the conflicts and addressed the comments you mentioned. Please have another look, thanks.

sairon reviewed (approved):
Thanks, looks good now, I just fixed some typos and adjusted raw formatting.
